packages/opentelemetry-plugin-http/test/utils/httpRequest.ts

```diff
@@ -18,54 +18,51 @@ import * as http from 'http';
 import * as url from 'url';
 import { RequestOptions } from 'https';
 
-export const httpRequest = {
-  get: (
-    options: string | RequestOptions
-  ): Promise<{
-    data: string;
-    statusCode: number | undefined;
-    resHeaders: http.IncomingHttpHeaders;
-    reqHeaders: http.OutgoingHttpHeaders;
-    method: string | undefined;
-  }> => {
-    const _options =
-      typeof options === 'string'
-        ? Object.assign(url.parse(options), {
-            headers: {
-              'user-agent': 'http-plugin-test',
-            },
-          })
-        : options;
-    return new Promise((resolve, reject) => {
-      const req = http.get(_options, (resp: http.IncomingMessage) => {
-        const res = (resp as unknown) as http.IncomingMessage & {
-          req: http.IncomingMessage;
-        };
-        let data = '';
-        resp.on('data', chunk => {
-          data += chunk;
-        });
-        resp.on('end', () => {
-          resolve({
-            data,
-            statusCode: res.statusCode,
-            /* tslint:disable:no-any */
-            reqHeaders: (res.req as any).getHeaders
-              ? (res.req as any).getHeaders()
-              : (res.req as any)._headers,
-            /* tslint:enable:no-any */
-            resHeaders: res.headers,
-            method: res.req.method,
-          });
-        });
-        resp.on('error', err => {
-          reject(err);
+type GetResult = Promise<{
+  data: string;
+  statusCode: number | undefined;
+  resHeaders: http.IncomingHttpHeaders;
+  reqHeaders: http.OutgoingHttpHeaders;
+  method: string | undefined;
+}>;
+
+function get(url: string | url.URL, options?: RequestOptions): GetResult;
+function get(url: RequestOptions): GetResult;
+function get(url: any, options?: any): GetResult {
+  return new Promise((resolve, reject) => {
+    let req: http.ClientRequest;
+    function getResponseCb(resp: http.IncomingMessage): void {
```

Comment:
Nit: `get` suggests that this callback should return something but it doesn't return anything.
WDYT about changing it to `responseCallback` or `onResponse` ?

Reply:
sorry, don't understand your comment. `get` returns a `Promise`.

Reply:
I think he is trying to point out the possible confusion with the name. Does `getResponseCb` mean "get a callback to be used for a response", or "callback used for the response from get". He is suggesting a name that removes that possible point of confusion.

Reply:
exactly the name `getResponseCb` is a bit misleading

Reply:
changed to `onGetResponseCb`